RUNTIME.md

```diff
@@ -0,0 +1,536 @@
+
+There are many ways to launch an MPI application, and there is no
+single uniform way of enforcing MPI rank CPU affinities across
+different job launch mechanisms.  Additionally, OpenMP runtimes, which
+are associated with the compiler choice, have different mechanisms for
+affinitizing OpenMP threads within CPUs available to each MPI process.
+To complicate things further the GEOPM control thread can be launched
+as an application thread or a process that may be part of the primary
+MPI application or a completely separate MPI application.  For these
+reasons it is difficult to document how to correctly affinitize
+processes in all configurations.  Please refer to your site
+documentation about CPU affinity for the best solution on the system
+you are using and consider extending the geopmlaunch wrapper to
+support your system configuration (please see the CONTRIBUTING.md file
+for information about how to share these implementation with the
+community).
+
+TESTING
+-------
+From within the source code directory, unit tests can be executed with
+the "make check" target.  The unit tests can be built without executing
+them with the "make checkprogs" target.  A typical parallel build and
+test cyle is executed with the following commands:
+
+    make -j
+    make checkprogs -j
+    make check
+
+The unit tests can be executed on any development system, including
+VMs and containers, that meets the BUILD REQUIREMENTS section above.
+
+The integration tests are located in the "integration/test" directory.
+These tests require a system meeting all of the requirements discussed
+in the RUN REQUIREMENTS section above.  The following script can be used
+to compile the GEOPM source, install into a specified location and execute
+the integration scripts in a work directory.
+
+    ##### USER SPECIFIC #####
+    # Set the location of the GEOPM source tree
+    export GEOPM_SOURCE=${HOME}/Git/geopm
+    # Set the install prefix for the geopm build
+    export GEOPM_INSTALL=${HOME}/build/geopm
+    # Set a place to locate the output of the tests
+    export GEOPM_WORKDIR=${HOME}/geopm_test
+    #####   ALL USERS   #####
+    # Execute out-of-place build and install using build script in base dir
+    ${GEOPM_SOURCE}/build.sh
+    # Set up run environment to use installed GEOPM
+    source $GEOPM_SOURCE/integration/config/run_env.sh
+    # Create a directory for output
+    mkdir -p $GEOPM_WORKDIR
+    cd $GEOPM_WORKDIR
+    # Run the integration tests
+    python3 $GEOPM_SOURCE/integration/test
+
+These integration tests are based on pyunit and leverage the geopmpy
+python package to validate the runtime.  Please report failures of
+these tests as issues.
+
+Note that if the `GEOPM_SKIP_COMPILER_CHECK` environment variable is
+set when `${GEOPM_SOURCE}/build.sh` is called, then the check to
+verify that the Intel compilers are used will be bypassed.  This can
+be useful for performing builds with the GNU toolchain.
+
+RESOURCE MANAGER INTEGRATION
```

Comment:
Are there implications of the service and libgeopmd.so that we have not captured in this section?

e.g. do we need libgeopmd.so to make these calls to save_control, etc.?